cnf-certification-test/accesscontrol/suite.go

```diff
@@ -644,28 +644,27 @@ func Test1337UIDs(env *provider.TestEnvironment) {
 // an allowed one will pass the test
 
 func testContainerSCC(env *provider.TestEnvironment) {
-	var badContainer []securitycontextcontainer.PodListcategory
-	var goodContainer []securitycontextcontainer.PodListcategory
+	var compliantObjects []*testhelper.ObjectOut
+	var nonCompliantObjects []*testhelper.ObjectOut
 	highLevelCat := securitycontextcontainer.CategoryID1
 	for _, pod := range env.Pods {
 		listCategory := securitycontextcontainer.CheckPod(pod)
 		for _, cat := range listCategory {
 			if cat.Category > securitycontextcontainer.CategoryID1NoUID0 {
-				badContainer = append(badContainer, cat)
+				aContainerOut := testhelper.NewContainerObjectOutBase(cat.NameSpace, cat.Podname, cat.Containername, "container category is NOT category 1 or category NoUID0", false).AddField(testhelper.Category, cat.Category.String())
```

Comment:
This is not a container. It looks like it needs a new type. Maybe, instead of creating plenty of very specific objects, we could use a "custom" type for objects different from Pod/Container/Process.

Reply:
Maybe a `pod/container/process` object func that returns a shared type of struct?

`func (p *Pod) ObjectOut(...params...) ObjectOut {}`

`func (c *Container) ObjectOut(...params...) ObjectOut {}`

Reply:
fixed. Used the SetType function to re-set the type a second time, while inheriting all the fields from the base type.